### PR TITLE
fix(DTFS2-6047): GEF coverStartDate coverEndDate validation

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue:
 * entering special characters on GEF facilities took you to problem page

# Changes made:
* Used joi to validate facility date inputs
* Added tests

### CODE REVIEW

File: gef-ui/server/controllers/about-facility/index.js [hunk 1]
```diff
@@ -116,21 +118,56 @@ const validateAboutFacility = async (req, res) => {
           subFieldErrorRefs: dateFieldsInError,
         });
       } else if (coverStartDateIsFullyComplete) {
+        let hasFormattingError = false;
         const now = new Date();
         const threeMonthsFromNow = add(now, { months: 3 });
+        // schema which ensures that coverStartDate year only contains 4 numbers
+        const yearSchema = Joi.string().length(4).pattern(/^[0-9]+$/).required();
```
Comment (IgnatG) [inline gef-ui/server/controllers/about-facility/index.js:125]: have a look at the SonarCloud messages around the regex pattern

File: gef-ui/server/controllers/about-facility/index.js [hunk 2]
```diff
@@ -69,6 +70,7 @@ const validateAboutFacility = async (req, res) => {
 
   let coverStartDate = null;
   let coverEndDate = null;
+  let coverEndDateValid = true;
 
   if (isTrueSet(body.hasBeenIssued)) {
```
Comment (IgnatG) [inline gef-ui/server/controllers/about-facility/index.js:75]: just to highlight that this function is big. really big... - around 300 lines of code. there is a lot of validation going on. 
after launch we need to split it in small chunks

Comment (Zainzzkk) [reply gef-ui/server/controllers/about-facility/index.js:75]: left todo

File: gef-ui/server/controllers/about-facility/index.js (hunk 1 above)
Comment (Zainzzkk) [reply gef-ui/server/controllers/about-facility/index.js]: done

**Review by IgnatG — APPROVED** [review]

**Pull request merged**